Dlw266/profile button and search bar functionality

danielwei15 opened the pull request:
### Summary <!-- Required -->

**Implemented the Profile Button graphically in its entirety.** 

This includes
- Collapse/Expand Arrows
- Profile Icon
- Username
- Edit, Manage, Favorited Club expanders, with Clubs underneath
- Simple transitions
- Sign out Button

All using dummy names for now.

![image](https://user-images.githubusercontent.com/31347660/157515954-9a0e17ae-dd15-4c24-9a76-6c3a99558534.png)

**Additionally, I also implemented searching in the searchbar:**

- The first part is to implement fetching club data from the backend (via a local fetch to localhost) and then to load that data into the clubs in ClubBoard.
- The second part is to implement a responsive search bar that can match user-entered strings to club names (as of now, exactly, without any fuzziness or searching anything else).

To accomplish this, I restructured the App a bit. Since the NavBar sometimes contains a search bar and sometimes doesn't, I created an interface for props for the component, with one optional (a callback function for when the user changes the searched string) and one required (a boolean for whether the NavBar should render a search bar at all). This NavBar should now be placed as a child component of other pages, such as the ClubBoard or Registration page, which can pass in the appropriate prop(s) according to the need of the page.

### Test Plan <!-- Required -->

No good test plans as it's purely graphical, but I tried to stick to the provided Figma for everything.

### Notes <!-- Optional -->

Didn't put in the correct font yet, as the rest of the site doesn't have the font yet either. If you want, I can implement that though.

Additionally, the main takeaway from the slight restructuring of the components is that the NavBar should now be a child component of other pages. This is already implemented, in this PR, for the existing components: ClubBoard and ClubRegistration.

### Breaking Changes  <!-- Optional -->

None at the moment.

danielwei15 committed:
Control State of Search Bar Input

danielwei15 committed:
Display the Club Data from Backend

danielwei15 committed:
Load Clubs on Component Mount

via useEffect

Also, Log ClubArray on Load from Backend

danielwei15 committed:
rewrite history

danielwei15 committed:
Merge branch 'dlw266/navbar-profile-button' into dlw266/new-main-for-merge

aaronkang7 reviewed (changes requested):
Great Job on this giant PR! I just wanted to point out that the drop-down appears behind the club cards on the dashboard! So you might want to check out the z-index in the css. Also, the signin button shouldn't have the dropdown button :). Other than that, great great great job! :D

danielwei15 committed:
fix requested changes

change z-index of dropdown to display on top of others; remove dropdown from signout button and add signout icon instead

aaronkang7 reviewed (approved):
LGTM!
